syntax = "proto3";

package tasks;

service TaskService {
    rpc CreateTask(CreateTaskRequest) returns (Task);
    rpc UpdateTask(UpdateTaskRequest) returns (Task);
    rpc DeleteTask(DeleteTaskRequest) returns (DeleteTaskResponse);
    rpc GetTaskById(GetTaskByIdRequest) returns (Task);
    rpc GetTasks(GetTasksRequest) returns (stream Task);
}

message CreateTaskRequest {
    string title = 1;
    string text = 2;
    string author = 3;
}

message UpdateTaskRequest {
    string task_id = 1;
    string title = 2;
    string text = 3;
    string updater = 4;
}

message DeleteTaskRequest {
    string task_id = 1;
    string user = 2;
}

message DeleteTaskResponse {
    bool success = 1;
}

message GetTaskByIdRequest {
    string task_id = 1;
    string user = 2;
}

message GetTasksRequest {
    string user = 1;
    int32 page_number = 2;
    int32 page_size = 3;
}

message Task {
    string task_id = 1;
    string title = 2;
    string text = 3;
    string author = 4;
}

